uniprotkb-rest/src/test/java/org/uniprot/api/uniprotkb/controller/UniProtKBSearchControllerIT.java

```diff
@@ -504,354 +507,478 @@ void canNotReturnFacetInformationForXML() throws Exception {
-                // given
-                UniProtKBEntry entry =
-                        UniProtEntryMocker.create(UniProtEntryMocker.Type.SP_CANONICAL);
-                getStoreManager().save(DataStoreManager.StoreType.UNIPROT, entry);
-
-                List<InactiveUniProtEntry> demergedList =
-                        InactiveEntryMocker.create(InactiveEntryMocker.InactiveType.DEMERGED);
-                getStoreManager()
-                        .saveEntriesInSolr(
-                                DataStoreManager.StoreType.INACTIVE_UNIPROT, demergedList);
-
-                // when search by accession field, returns only itself
-                ResultActions response =
-                        getMockMvc()
-                                .perform(
-                                        get(SEARCH_RESOURCE + "?query=accession:Q00007")
-                                                .header(ACCEPT, APPLICATION_JSON_VALUE));
-
-                response.andDo(log())
-                        .andExpect(status().is(HttpStatus.OK.value()))
-                        .andExpect(
-                                header().string(HttpHeaders.CONTENT_TYPE, APPLICATION_JSON_VALUE))
-                        .andExpect(jsonPath("$.results.size()", is(1)))
-                        .andExpect(jsonPath("$.results.*.primaryAccession", contains("Q00007")))
-                        .andExpect(jsonPath("$.results.*.entryType", contains("Inactive")))
-                        .andExpect(
-                                jsonPath(
-                                        "$.results.*.inactiveReason.inactiveReasonType",
-                                        contains("DEMERGED")))
-                        .andExpect(
-                                jsonPath(
-                                        "$.results.*.inactiveReason.mergeDemergeTo",
-                                        contains(contains("P21802", "P63151"))));
-
-                // when search accession by default field, returns only itself
-                response =
-                        getMockMvc()
-                                .perform(
-                                        get(SEARCH_RESOURCE + "?query=Q00007")
-                                                .header(ACCEPT, APPLICATION_JSON_VALUE));
-
-                response.andDo(log())
-                        .andExpect(status().is(HttpStatus.OK.value()))
-                        .andExpect(
-                                header().string(HttpHeaders.CONTENT_TYPE, APPLICATION_JSON_VALUE))
-                        .andExpect(jsonPath("$.results.size()", is(1)))
-                        .andExpect(jsonPath("$.results.*.primaryAccession", contains("Q00007")));
-            }
+    @ParameterizedTest
+    @Tag("TRM_27083")
+    @ValueSource(strings = {"reviewed", "unreviewed"})
+    void inactiveEntry_canSearchForFullIDs(String type) throws Exception {
```

Comment:
Edd, Do you think we could use InactiveEntryMocker to add inactive entries? I think  (I might be wrong) there is no inactive entry in this test.
I think the same thing is happening with other tests.

Reply:
Gosh -- I named the test wrong. You are right -- it's not related to inactive entries. But is for https://www.ebi.ac.uk/panda/jira/browse/TRM-27083 and all the conditions required by Elisabeth/Maria, etc. I'll rename the test.